addons/pos_restaurant/models/pos_order.py

```diff
@@ -140,7 +141,7 @@ def get_table_draft_orders(self, table_id):
 
         for order in table_orders:
             order['pos_session_id'] = order['session_id'][0]
-            order['uid'] = order['pos_reference'].split(' ')[1]
+            order['uid'] = search(r"\d{5}-\d{3}-\d{4}", order['pos_reference']).group(0)
```

Comment:
Are you sure the size never changes? Maybe it is enough to keep just `\d`

Reply:
The size is hard-coded here: https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/static/src/js/models.js#L2493

Reply:
It is a zero pad so if you make >9999 orders in one day, you get 5 numbers at the end. I think I already got the case of a big shop getting over one of the limit (I don't recall which one).